miden-tx/src/testing/mock_chain/mod.rs

```diff
@@ -391,135 +379,145 @@ impl MockChain {
+            .init_seed(self.rng.gen())
+            .nonce(Felt::ONE)
+            .with_component(BasicWallet)
+            .with_assets(assets);
+        self.add_from_account_builder(auth_method, account_builder)
+    }
+
+    /// Adds a new faucet with the specified authentication method and metadata.
+    pub fn add_new_faucet(
+        &mut self,
+        auth_method: Auth,
+        token_symbol: &str,
+        max_supply: u64,
+    ) -> MockFungibleFaucet {
+        let account_builder = AccountBuilder::new()
+            .init_seed(self.rng.gen())
+            .nonce(Felt::ZERO)
+            .account_type(AccountType::FungibleFaucet)
+            .with_component(
+                BasicFungibleFaucet::new(
+                    TokenSymbol::new(token_symbol).unwrap(),
+                    10,
+                    max_supply.try_into().unwrap(),
+                )
+                .unwrap(),
+            );
 
-        self.add_from_account_builder(account_builder, Auth::BasicAuth(auth_secret_key))
+        MockFungibleFaucet(self.add_from_account_builder(auth_method, account_builder))
     }
 
-    /// Helper method to build a faucet account
-    fn add_faucet_account(
+    /// Adds an existing (nonce == 1) faucet with the specified authentication method and metadata.
+    pub fn add_existing_faucet(
         &mut self,
-        token_symbol_str: &str,
+        auth_method: Auth,
+        token_symbol: &str,
         max_supply: u64,
         total_issuance: Option<u64>,
-        nonce: Felt,
     ) -> MockFungibleFaucet {
-        let token_symbol = TokenSymbol::new(token_symbol_str).unwrap();
-
-        let fungible_faucet = "export.::miden::contracts::faucets::basic_fungible::distribute
-        export.::miden::contracts::faucets::basic_fungible::burn
-        export.::miden::contracts::auth::basic::auth_tx_rpo_falcon512";
+        // Note: data is stored as [a0, a1, a2, a3] but loaded onto the stack as
+        // [a3, a2, a1, a0, ...]
+        let metadata = [
+            Felt::new(max_supply),
+            Felt::new(10u64),
+            TokenSymbol::new(token_symbol).unwrap().into(),
+            Felt::ZERO,
+        ];
+
+        let faucet_component = AccountComponent::new(basic_fungible_faucet_library(), vec![StorageSlot::Value(metadata)])
+            .expect("basic fungible faucet component should satisfy the requirements of a valid account component")
+            .with_supported_type(AccountType::FungibleFaucet);
+
+        let account_builder = AccountBuilder::new()
+            .with_component(faucet_component)
+            .nonce(Felt::ONE)
+            .init_seed(self.rng.gen())
+            .account_type(AccountType::FungibleFaucet);
 
-        let code =
-            AccountCode::compile(fungible_faucet, TransactionKernel::testing_assembler(), true)
-                .unwrap();
+        let (mut account, seed, authenticator) = match auth_method {
+            Auth::BasicAuth => {
```

Comment:
I forgot to do a second pass on this function. I mostly copied code because I didn't find a way to re-set the first slot (for the total issuance) in a way that fit the context of the function (ie, from `AccountBuilder` before passing it on to `add_from_account_builder()`)